src/test/librbd/io/test_mock_CopyupRequest.cc

```diff
@@ -26,6 +26,10 @@ struct MockTestImageCtx : public MockImageCtx {
     : MockImageCtx(image_ctx) {
     parent = mock_parent_image_ctx;
   }
+  ~MockTestImageCtx() override {
+    // copyups need to complete prior to attempting to delete this object
+    wait_for_async_ops();
```

Comment:
Note: thinking to myself, I wonder if this might be (partially) responsible for some of the sporadic `unittest_librbd` failures since it looks like some of the test cases don't invoke `flush_async_operations` which was used to ensure the copy-ups were completed before destroying this object.